src/Sulu/Bundle/AdminBundle/Navigation/DataNavigationItem.php

```diff
@@ -0,0 +1,267 @@
+<?php
+/*
+ * This file is part of the Sulu CMF.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Navigation;
+
+/**
+ * Represents an item in the navigation with data-navigation as "third layer".
+ */
+class DataNavigationItem extends NavigationItem
+{
+    /**
+     * Url to load data from
+     * @var string
+     */
+    protected $dataUrl;
+
+    /**
+     * Key of the result array
+     * @var string
+     */
+    protected $dataResultKey;
+
+    /**
+     * Key of the name of entity
+     * @var string
+     */
+    protected $dataNameKey;
+
+    /**
+     * Key of the children link
+     * @var string
+     */
+    protected $dataChildrenLinkKey;
```

Comment:
Forget about it, saw the later code, would probably quite confusing without the prefix.